### PR TITLE
Organize helper functions from the `service` module

### PR DESCRIPTION
**Pull request by fredmorcos**
This takes some common tasks whose functions were lurking in the `imdb::service` module and moves them under the more generic `utils` module. There are a few renames as well.

### CODE REVIEW

File: lib/src/utils/io/file.rs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+use crate::utils::result::Res;
+
+/// Returns the file at the given path if it exists, or an Ok Result if it is not found.
+///
+/// Only returns an error if a problem occurs while opening an existing file.
+///
+/// # Arguments
+///
+/// * `path` - Path of the file to be opened.
+pub fn exists(path: &Path) -> Res<Option<File>> {
```
Comment (HenningHolmDE) [inline lib/src/utils/io/file.rs:19]: Maybe it's just me, but I find a function named `exists` returning a file handle (and not just a boolean) a bit strange. 
Could `open_existing` be a better alternative?

File: lib/src/utils/io/file.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+pub fn read_static(filename: &Path) -> Res<&'static [u8]> {
+  Ok(Box::leak(fs::read(filename)?.into_boxed_slice()))
+}
+
+/// Creates a file and returns a buffered writer handle to it.
+///
+/// # Arguments
+///
+/// * `file` - The file path to create.
+pub fn make_buffered(filename: &Path) -> Res<BufWriter<File>> {
```
Comment (HenningHolmDE) [inline lib/src/utils/io/file.rs:65]: Maybe `create_buffered` would be clearer for this? (as is wraps `File::create`)

**Review by HenningHolmDE — APPROVED** [review]

File: lib/src/utils/io/file.rs (hunk 1 above)
Comment (fredmorcos) [reply lib/src/utils/io/file.rs:19]: Agreed, `open_existing` is clearer.

**Comment by fredmorcos** [comment]
I also added a commit that cleanups up some documentation strings.

**Review by HenningHolmDE — APPROVED** [review]

**Pull request merged**